jupyter_accessibility/project.py

```diff
@@ -0,0 +1,428 @@
+    )
+    links: Path = Field(
+        Path("yarn-links"),
+        description="the folder we use to hold our yarn symlinks for development assets",
+    )
+    prefix: Path = Field(None, description="a computed location for our conda environment")
+    # this is why we don't need to nox, this exposes more knobs for special cases.
+    conda: Union[bool, str] = Field(
+        None, description="the conda run prefix used to execute isolated system commands"
+    )
+    # with this we can turn conda, node, and pip in offline mode
+    offline: bool = Field(False, description="build in offline mode using your caches")
+    yarn_links: Path = Field(None, description="the computed location of our yarn symbollic links")
+    test_results: Path = Field(
+        None, description="the computed location of our playwright test results"
+    )
+
+    def __post_init__(self):
+        # compute the prefix and carry the absolute position forward
+        self.prefix = (self.dir / self.env).resolve()
+
+        # compute our conda run statement that prepends all the calls we make
+        if self.conda is None:
+            self.conda = f"conda run --no-capture-output --prefix {self.prefix}"
+            if self.offline:
+                self.conda += " --offline"
+        elif not self.conda:
+            # the empty string case is the escape hatch from using conda.
+            # under these conditions you are using the environment you are in
+            self.conda = ""
+        # compute the location of the yarn links we manage
+        self.yarn_links = (self.dir / self.links).resolve()
+
+        # the output location of the test results for easy access
+        self.test_results = self.dir / AXE_TEMPLATE.name / "test-results"
+
+        # initialize each of the Package/Repo objects that contain
+        # instructions for executing things
+        for id in self.ids:
+            kwargs = {}
+            if "://" in id:
+                # currently, everything at a url is assumed to be a repo
+                # there are certainly causes where we are wrong. we might
+                # likely need adapters for different network locations.
+
+                if "@" in id:
+                    id, _, branch = id.rpartition("@")
+                    kwargs.setdefault("branch", branch)
+                repo = Repo(url=id, parent=self, **kwargs)
+            else:
+                # otherwise we assume a built package.
+                repo = Package(name=id, parent=self)
+
+            self.repos[repo.name] = repo
+
+    # this is a setup task and the description could be improved
+    def task_env(self):
+        """create a conda environment for development work"""
+        yield self.conda and dict(
+            name="conda",
+            actions=[
+                do(
+                    f'conda create -yc conda-forge --prefix {self.prefix} python=3.9 "nodejs>=14,<15" yarn git'
```

Comment:
This can do for now but might need to devise a way to not hardcode / pin versions here